{
  if (e.getCause() instanceof IOException)   throw (IOException)e.getCause();
  throw new RuntimeException(e);
}
